test/models/user_test.rb

```diff
@@ -762,6 +762,17 @@ def setup_user(operation, type = 'users', search = nil)
         end
       end
 
+      test 'existing user of different authsource' do
+        assert_difference('User.count', 0) do
+          created_user = User.find_or_create_external_user(
+            {:login => @user.login,
```

Comment:
Perhaps use one of the fixtures to make it clear that the user already exists?
```suggestion
            {:login => users(:one).login,
```